feat:

MaXM91 opened the pull request:
- добавить package model c User и Film;
- добавить описание пользователя(User.class);
- добавить описание фильма(Film.class);
- добавить package с UserController и FilmController;
- добавить контроллер пользователей (Хранилише по ид и обработчик запросов UserController.class);
- добавить контроллер фильмов (Хранилище по ид и обработчик запросов FilmController.class);
- добавить обработчики событий IdNotNullException, ValidationException, MinimumDateValidator;

Commit:
feat:
- добавить package model c User и Film;
- добавить описание пользователя(User.class);
- добавить описание фильма(Film.class);
- добавить package с UserController и FilmController;
- добавить контроллер пользователей (Хранилише по ид и обработчик запросов UserController.class);
- добавить контроллер фильмов (Хранилище по ид и обработчик запросов FilmController.class);
- добавить обработчики событий IdNotNullException, ValidationException, MinimumDateValidator;

pattyku commented on pom.xml:
```diff
@@ -0,0 +1,54 @@
+	<version>0.0.1-SNAPSHOT</version>
+	<name>filmorate</name>
+	<description>Demo project for Spring Boot</description>
+	<properties>
```
Очень не хватает отделения друг от друга (пустыми строками) глобальных частей таких как parent, properties, dependencies

Ну и как минимум не хватает указания кодировки в properties
<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>

Кроме этого, советую всем соблюдать некий порядок при объявлении зависимостей.
Сначала "главный" spring все его зависимости и всё о нём.
Потом по степени "тяжеловестности" остальные зависимости.

pattyku commented on src/main/java/ru/yandex/practicum/filmorate/FilmorateApplication.java:
```diff
@@ -0,0 +1,13 @@
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class FilmorateApplication {
```
Тут все отлично)
Любимый класс для проверки

pattyku commented on src/main/java/ru/yandex/practicum/filmorate/controllers/FilmController.java:
```diff
@@ -0,0 +1,52 @@
+
+    @PostMapping
+    public Film addFilm(@RequestBody @Valid Film film) throws ValidationException, IdNotNullException {
+        if (new ArrayList<>(films.values()).stream().anyMatch(u -> u.getName().equals(film.getName())) && !films.isEmpty()) {
```
Не нужна такая проверка при создании фильма. Могут быть фильмы с одинаковым названием

pattyku commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,30 @@
+    @Size(max = 200)
+    private String description;
+
+    @MinimumDate
```
Тут отлично, используешь аннотации, еще и свою создал

pattyku commented on src/main/java/ru/yandex/practicum/filmorate/validators/exceptions/IdNotNullException.java:
```diff
@@ -0,0 +1,7 @@
+package ru.yandex.practicum.filmorate.validators.exceptions;
+
+public class IdNotNullException extends Throwable {
```
extends RuntimeException
от  Throwable практически никогда не наследуемся. Или  RuntimeException или  Exception

pattyku commented on src/main/java/ru/yandex/practicum/filmorate/validators/exceptions/ValidationException.java:
```diff
@@ -0,0 +1,7 @@
+package ru.yandex.practicum.filmorate.validators.exceptions;
+
+public class ValidationException extends Throwable {
```
extends RuntimeException

pattyku commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,25 @@
+    private String email;
+
+    @NotEmpty
+    @NotBlank
```
Что-то не нашла по коду где проверяешь пробел в логине
Можно сделать аннотацией
@Pattern(regexp = "\\S*", message = "Логин не может содержать пробелы.")

Commit:
delete:
- удалить проверку на повтор названия фильма в FilmController;
- изменить наследование в IdNotNullException и ValidationException на RuntimeException;
- отделить разделы пробелами POM.xml.
